planner/plansdk/plansdk.go

```diff
@@ -122,32 +122,51 @@ func MergePlans(plans ...*Plan) *Plan {
 	plan := &Plan{
 		DevPackages:     []string{},
 		RuntimePackages: []string{},
-		SharedPlan: SharedPlan{
-			InstallStage: &Stage{},
-			BuildStage:   &Stage{},
-			StartStage:   &Stage{},
-		},
 	}
 	for _, p := range plans {
-		err := mergo.Merge(plan, p, mergo.WithAppendSlice)
+		err := mergo.Merge(
+			plan,
+			&Plan{
+				DevPackages:     p.DevPackages,
+				RuntimePackages: p.RuntimePackages,
+			},
+			// Only WithAppendSlice the dev and runtime packages field.
+			mergo.WithAppendSlice,
+		)
 		if err != nil {
 			panic(err) // TODO: propagate error.
 		}
 	}
 
 	plan.DevPackages = pkgslice.Unique(plan.DevPackages)
 	plan.RuntimePackages = pkgslice.Unique(plan.RuntimePackages)
+	plan.SharedPlan = PickSharedPlan(plans...)
 
-	// Set default files for install stage to copy.
-	if plan.SharedPlan.InstallStage.InputFiles == nil {
-		plan.SharedPlan.InstallStage.InputFiles = []string{"."}
+	return plan
+}
+
+func PickSharedPlan(plans ...*Plan) SharedPlan {
+	for _, p := range plans {
+		// For now, pick the first buildable plan.
+		if p.Buildable() {
+			return p.SharedPlan
+		}
 	}
-	// Set default files for install stage to copy over from build step.
-	if plan.SharedPlan.StartStage.InputFiles == nil {
-		plan.SharedPlan.StartStage.InputFiles = []string{"."}
+	return SharedPlan{}
+}
+
+func OverrideSharedPlan(dst *Plan, src *Plan) SharedPlan {
```

Comment:
Having a separate function we need to call in addition to MergePlans can lead to mistakes later on (if we forget to call it).

Thoughts on changing interface to:

`MergeAutomatedPlans` (or some similar name, designed to merge plans that come from planners.

`MergeBasePlan(automatedPlan, basePlan)`  (designed to merge json plan into automated one)


Sorry for all naming nits, I'm just getting confused with the planner plan naming and shared plan naming.